### PR TITLE
refactor: Breadcrumb separator use `li` instead `span`

### PR DESCRIPTION
**Pull request by heiyu4585**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution
https://github.com/ant-design/ant-design/pull/40867#issuecomment-1439365589

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Unify the `span` in `ul` to `li` to make the DOM structure more standardized. Please pay attention to the problems of missing overlay styles and element acquisition    |
| 🇨🇳 Chinese |     规范 Breadcrumb 分隔符的 DOM 结构，从 span 统一为 li 元素，请注意覆盖样式丢失和元素获取的问题     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/breadcrumb/BreadcrumbItem.tsx [hunk 1]
```diff
@@ -85,10 +85,14 @@ const BreadcrumbItem: CompoundedComponent = (props) => {
...
-      <li>
-        {link}
-        {separator && <span className={`${prefixCls}-separator`}>{separator}</span>}
-      </li>
+      <>
+        <li>{link}</li>
+        {separator && (
+          <li>
+            <span className={`${prefixCls}-separator`}>{separator}</span>{' '}
+          </li>
```
Comment (afc163) [inline components/breadcrumb/BreadcrumbItem.tsx:93]: 内部也是不是直接用 Breadcrumb.Separator 实现？

**Comment by afc163** [comment]
dom 结构变化，发到 feature 分支

File: components/breadcrumb/BreadcrumbItem.tsx (hunk 1 above)
Comment (heiyu4585) [reply components/breadcrumb/BreadcrumbItem.tsx:93]: ok

**Comment by heiyu4585** [comment]
> dom 结构变化，发到 feature 分支

ok

File: components/breadcrumb/BreadcrumbItem.tsx (hunk 1 above)
Comment (heiyu4585) [reply components/breadcrumb/BreadcrumbItem.tsx]: 我正好在做 组件的 转 data,我在那里考虑呢 @afc163 @zombieJ 
https://github.com/ant-design/ant-design/pull/40543

File: components/breadcrumb/BreadcrumbItem.tsx [hunk 2]
```diff
@@ -85,10 +86,10 @@ const BreadcrumbItem: CompoundedComponent = (props) => {
   link = renderBreadcrumbNode(link);
   if (children !== undefined && children !== null) {
     return (
-      <li>
-        {link}
-        {separator && <span className={`${prefixCls}-separator`}>{separator}</span>}
-      </li>
+      <>
+        <li>{link}</li>
+        {separator && <BreadcrumbSeparator>{children}</BreadcrumbSeparator>}
```
Comment (afc163) [inline components/breadcrumb/BreadcrumbItem.tsx:91]:
```suggestion
        {separator && <BreadcrumbSeparator>{separator}</BreadcrumbSeparator>}
```

Comment (heiyu4585) [reply components/breadcrumb/BreadcrumbItem.tsx]: done

**Comment by afc163** [comment]
PR 上的 changelog 里写一下对用户的可能影响。

**Comment by afc163** [comment]
类似 https://4x.ant.design/changelog-cn#4.3.0 里 tabs 这个写法。

**Comment by heiyu4585** [comment]
> 类似 https://4x.ant.design/changelog-cn#4.3.0 里 tabs 这个写法。

done

**Review by afc163 — APPROVED** [review]

**Pull request merged**